File: src/main/java/com/campool/controller/AdminController.java
package com.campool.controller;

import com.campool.model.AdminSignUp;
import com.campool.service.AdminService;
import javax.validation.Valid;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RestController;

@RequiredArgsConstructor
@RestController
public class AdminController {

    @NonNull
    private final AdminService adminService;

    @PostMapping("/admins")
    public void signUpUser(@Valid AdminSignUp adminSignUp) {
        adminService.add(adminSignUp);
    }

}

File: src/main/java/com/campool/controller/AdminController.kt
package com.campool.controller

import com.campool.annotation.LoginValidation
import com.campool.enumeration.Role
import com.campool.model.AdminLoginRequest
import com.campool.model.AdminSignUpRequest
import com.campool.service.AdminService
import com.campool.service.AuthService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RestController
import javax.validation.Valid

@RestController
class AdminController(
    private val adminService: AdminService,
    private val authService: AuthService
) {
    @PostMapping("/admins")
    fun signUpAdmin(@Valid adminSignUp: AdminSignUpRequest) {
        adminService.add(adminSignUp)
    }

    @PostMapping("/admins/login")
    fun loginAdmin(@Valid request: AdminLoginRequest) {
        authService.authenticate(request)
    }

    @LoginValidation(role = Role.ADMIN)
    @GetMapping("/admins/logout")
    fun logoutAdmin() {
        authService.deauthenticate()
    }
}